kernel/mem_domain.c

```diff
@@ -13,6 +13,17 @@
 static u8_t max_partitions;
 
 
+static void ensure_w_xor_x(u32_t attrs) {
+#if defined(CONFIG_EXECUTE_XOR_WRITE)
+	bool writable = K_MEM_PARTITION_IS_WRITABLE(attrs);
+	bool executable = K_MEM_PARTITION_IS_EXECUTABLE(attrs);
+
+	__ASSERT(writable != executable);
```

Comment:
does this compile? I think __ASSERT_NO_MSG is what you are looking for, __ASSERT takes a string as well as the expression

Reply:
think you might also need #include <misc/__assert.h> unless kernel.h or something pulls it in

Reply:
Done.